### PR TITLE
Add part 3 cqrs/es dice game post

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

File: _posts/27-07-2015-event-sourced-dice-game-part3.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+- catch events
+- update and store the statistics
+- expose REST API
+
+I'll focus on each of these in the corresponding sections below.
+
+## <a name="catching-events" href="#catching-events">Catching events</a>
+
+Similar to the webapp part, here, once again, we'll use Reactive rabbit to consume the events from RabbitMQ.
```
Comment (leszekgruchala) [inline _posts/27-07-2015-event-sourced-dice-game-part3.md]: `Reactive rabbit to` -> `Reactive Rabbit to`

File: _posts/27-07-2015-event-sourced-dice-game-part3.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+We'll create another queue and bind it to the events exchange:
+
+<script src="https://gist.github.com/LukasGasior1/8bcf66e19ac388ae7104.js"></script>
+[\[StatsApp.scala\]](https://github.com/LukasGasior1/event-sourced-dice-game/blob/master/statistics/src/main/scala/lgasior/dicegame/statistics/StatsApp.scala)
+
+The `"x-match" -> "all", "type" -> "DiceRolled"` bind arguments mean: "I want to receive `DiceRolled` events from all played games".
```
Comment (leszekgruchala) [inline _posts/27-07-2015-event-sourced-dice-game-part3.md]: `who is given a reference to` -> `which is given a reference to s`

`Who` is used only to people so far I know.

**Comment by leszekgruchala** [comment]
Big :clap:

File: _posts/27-07-2015-event-sourced-dice-game-part3.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+---
+
+Welcome to the last part of dice game post series!
+
+We already have fully working game that we created in [1st](http://blog.scalac.io/2015/03/20/event-sourced-dice-game-part1.html) and [2nd](http://blog.scalac.io/2015/06/11/event-sourced-dice-game-part2.html) parts.
+
+Today we'll see how we can take advantage of the fact that we're using event sourcing.
+
+We'll create a separate statistics project that will catch all `DiceRolled` events and
+count how many times each of the possible numbers have been rolled.
```
Comment (pjazdzewski1990) [inline _posts/27-07-2015-event-sourced-dice-game-part3.md]: maybe simply `each number ...`? I can imagine we are not counting numbers impossible to roll ;)

File: _posts/27-07-2015-event-sourced-dice-game-part3.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+Now it's time to take a look at the `StatsActor` itself.
+
+It's a `PersistentActor` that can handle `IncRollsCount` command.
+Once it receives it, `RollsCountIncreased` event is generated and saved.
+
+<script src="https://gist.github.com/LukasGasior1/253cd74fcdb53770460b.js"></script>
+[\[StatsActor.scala\]](https://github.com/LukasGasior1/event-sourced-dice-game/blob/master/statistics/src/main/scala/lgasior/dicegame/statistics/actor/StatsActor.scala)
+
+Here, in contrast to [`GameActor`](https://github.com/LukasGasior1/event-sourced-dice-game/blob/master/game/src/main/scala/lgasior/dicegame/actor/GameActor.scala) from [1st part](http://blog.scalac.io/2015/03/20/event-sourced-dice-game-part1.html), `persistenceId` is constant as we'll have only one instance of `StatsActor` ever (created when the application starts).
```
Comment (pjazdzewski1990) [inline _posts/27-07-2015-event-sourced-dice-game-part3.md]: Putting GameActor in ticks obfuscates the fact that it's a link ... I would do: in contract to `GameActor` ([link](link to repo))

File: _posts/27-07-2015-event-sourced-dice-game-part3.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+First, let's run everything we need, i.e.:
+
+- RabbitMQ
+- game backend server
+- game frontend app
+- our shiny statistics app
+
+Here are the commands I use:
+
+<script src="https://gist.github.com/LukasGasior1/ebfb1eb07442f9d490c7.js"></script>
```
Comment (pjazdzewski1990) [inline _posts/27-07-2015-event-sourced-dice-game-part3.md]: Can we drop the triangles? They don't bring much benefit, but make copy-paste much harder

File: _posts/27-07-2015-event-sourced-dice-game-part3.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+If everything went good, we should now have all services running.
+
+What does our statistics service return? Let's check that...
+
+<script src="https://gist.github.com/LukasGasior1/497b70e5b80ba9e0768a.js"></script>
```
Comment (pjazdzewski1990) [inline _posts/27-07-2015-event-sourced-dice-game-part3.md]: It would be good if you could add `// Output` between command and output

**Comment by pjazdzewski1990** [comment]
:+1:

**Pull request merged**